src/cron/syncSubscription.ts

```diff
@@ -0,0 +1,29 @@
+import { syncSubscription } from '../common';
+import { StorageKey, StorageTopic, generateStorageKey } from '../config';
+import { logger } from '../logger';
+import { getRedisListLength, popFromRedisList } from '../redis';
+import { Cron } from './cron';
+
+const redisKey = generateStorageKey(
+  StorageTopic.CIO,
+  StorageKey.Reporting,
+  'global',
+);
+
+const cron: Cron = {
+  name: 'sync-subscription',
+  handler: async (con) => {
+    logger.info('syncing subscriptions');
+
+    while ((await getRedisListLength(redisKey)) > 0) {
```

Comment:
Can we make this finite somehow, `while` loops are always scary.

Reply:
Added a bit more robustness to break out of the loop. Is that alright?

Reply:
What did you add, I don't see anything? (might be github lagging)

Reply:
Forgot to push 🙈

Reply:
Now they're pushed